test/tstDetailsHalfTraversal.cpp

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborXTest_StdVectorToKokkosView.hpp"
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include "ArborX_EnableViewComparison.hpp"
+#include <ArborX_DetailsHalfTraversal.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Random.hpp>
+
+#include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
+
+namespace Test
+{
+template <class ExecutionSpace>
+// enclosing parent function for an extended __host__ __device__ lambda must not
+// have deduced return type
+#ifndef KOKKOS_COMPILER_NVCC
+auto
+#else
+Kokkos::View<ArborX::Point *, ExecutionSpace>
+#endif
```

Comment:
We have a ton of places in ArborX where we would need this comment. I'd say don't do it here, rather just explicitly state the return type. If nvcc ever supports it properly, we'll have to change everywhere anyway.